### PR TITLE
Add location tracking to Atomic types

### PR DESCRIPTION
**Pull request by seanmonstar**

### CODE REVIEW

File: src/rt/location.rs [hunk 1]
```diff
@@ -103,6 +103,9 @@ impl PanicBuilder {
         if let Some(width) = width {
             msg = format!("\n{}", msg);
             for (key, thread, location) in &self.locations {
+                if !location.is_captured() {
```
Comment (seanmonstar) [inline src/rt/location.rs]: Note: I saw a subtraction overflow sometimes here.

File: src/sync/atomic/atomic.rs [hunk 2]
```diff
@@ -18,43 +18,51 @@ where
         Atomic { state }
     }
 
+    #[cfg_attr(loom_nightly, track_caller)]
```
Comment (carllerche) [inline src/sync/atomic/atomic.rs]: I don't think you want to track the caller here. Instead, you probably want to update these fns to take `Location` and track the caller on the wrappers (`AtomicPtr`, `AtomicBool`, `Atomic*`, ..).

Comment (hawkw) [reply src/sync/atomic/atomic.rs]: Yeah, I think that the caller here is always going to be inside loom?

Comment (seanmonstar) [reply src/sync/atomic/atomic.rs]: I don't believe it matters either way, the location is does get correctly propagated to the user's call on `AtomicPtr::whatever()` (visually tested).

**Pull request merged**